app/src/test/java/com/codesoom/assignment/application/TaskServiceTest.java

```diff
@@ -0,0 +1,200 @@
+package com.codesoom.assignment.application;
+
+import com.codesoom.assignment.TaskNotFoundException;
+import com.codesoom.assignment.models.Task;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+
+import java.util.List;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+@DisplayName("TaskService 클래스")
+class TaskServiceTest {
+
+    private TaskService taskService;
+
+    private static final String TASK_TITLE = "test";
+    private static final String UPDATE_POSTFIX = "!!!";
+    private static final Long VALID_ID = 1L;
+    private static final Long INVALID_ID = 0L;
+
+    @BeforeEach
+    void setUp() {
+        taskService = new TaskService();
+
+        Task task = new Task();
+        task.setTitle(TASK_TITLE);
+
+        taskService.createTask(task);
+    }
+
+    @Nested
+    @DisplayName("getTask 메소드는")
+    class Describe_getTask {
+        @Test
+        @DisplayName("등록된 Task 전체 리스트를 리턴한다.")
+        void it_return_tasks() {
+            List<Task> tasks = taskService.getTasks();
+            Task foundTask = tasks.get(0);
+
+            assertThat(tasks).hasSize(1);
+            assertThat(foundTask.getTitle()).isEqualTo(TASK_TITLE);
+        }
+
+        @Nested
+        @DisplayName("만약 등록된 Task의 id 값이 주어진다면")
+        class Context_with_id {
+            @Test
+            @DisplayName("등록된 task 정보를 리턴한다.")
+            void it_return_task() {
+                Task foundTask = taskService.getTask(VALID_ID);
+
+                assertThat(foundTask.getTitle()).isEqualTo(TASK_TITLE);
+            }
+        }
+
+        @Nested
+        @DisplayName("만약 등록되지 않은 Task의 id 값이 주어진다면")
+        class Context_with_invalid_id {
+            @Test
+            @DisplayName("TaskNotFoundException 을 리턴한다.")
```

Comment:
표현을 꼼꼼히 하면 좋습니다.
```suggestion
            @DisplayName("Task를 찾을 수 없다는 예외를 던진다.")
```
```suggestion
            @DisplayName("TaskNotFoundException를 던진다.")
```

왜 "던진다"라는 표현을 쓰는지에 대해서도 생각해보세요. "예외를 리턴한다"는 표현이 정확한 경우는 딱 한 가지 밖에 없습니다. 그 메소드의 리턴 타입이 정말 그 예외 타입인 경우에요.

말하자면 이런 경우죠.
```java
public TaskNotFoundException getTask(long id) {
```

하지만 리턴 타입은 이렇지 않죠?
실제로는 이럴 것입니다.

```java
public Task getTask(long id) throws TaskNotFoundException {
```

물론 여기에서 `throws` 부터는 생략하셨을 가능성도 있어요. 하지만 여기에서 생략했는지 안했는지는 중요하지 않습니다.
메소드 시그니처에서 예외 앞에 어떤 단어를 두는지 눈여겨보세요. `throws`가 있습니다.
예외는 던지는 것이라고 자바 언어는 표현하고 있는 것이에요.

Reply:
아하!! 맞네요. 
return : 리턴한다. 
throws : 던진다. 

예외 처리할 때 자주 쓰는데도 한글로 어떻게 표현할지에 대해 생각해보지 않으니 새롭게 다가오네요 ..ㅎㅎ 
쓰는 단어에 대해 눈여겨보겠습니다!!